containers/containers.cabal

```diff
@@ -24,6 +24,7 @@ cabal-version:  >=1.10
 extra-source-files:
     include/containers.h
     changelog.md
+    mkappend.hs
```

Comment:
I'm not familiar with this section of the Cabal file. What is the purpose of adding the file there?

Reply:
It is a list of additional files to be included in source distributions -- figured it made sense to include here, though won't really effect many users if we didn't include it

Reply:
Ah, okay. So ... stuff that you can download using `cabal-install`?

Reply:
I believe it'll cause it to be included in the tarball uploaded to/downloaded from Hackage, yes. Tbh my confidence is somewhat low on my understanding of this, just bumped up against it a few times when messing with my own packages a while ago.

Reply:
I guess that's also means people will be able to browse it on Hackage, which is good.

Reply:
Actually, not sure you can actually view this list of files in the UI. Would probably make sense for Hackage to show them, but an example I used it for was a compiled frontend for a web package, so maybe there are some packages for which having the whole index wouldn't render very nicely.

Reply:
Won't they show up with all the rest of the files in [this listing](https://hackage.haskell.org/package/containers-0.6.7/src/)?

Reply:
Oh neat, never saw that. Thanks!

Reply:
Search for "Cabal source package" on the main Hackage page of any package and click on "browse". It's kind of buried, but it's there, and useful.